java/server/src/org/openqa/grid/web/servlet/api/v1/Proxy.java

```diff
@@ -39,7 +41,11 @@
     }
 
     final String proxyToFind = query.replaceAll("^/", "");
-    final RemoteProxy proxy = this.getRegistry().getAllProxies().getProxyById(proxyToFind);
+    RemoteProxy proxy = this.getRegistry().getAllProxies().getProxyById(proxyToFind);
```

Comment:
> There are use cases where 'autodetecting' or 'assuming' the proxy id is not sufficient.

@mach6 - Can you please help call out what those use cases might be ? I would be curious to learn/hear about them